testing/simapp/test_helpers.go

```diff
@@ -234,6 +234,8 @@ func CheckBalance(t *testing.T, app *SimApp, addr sdk.AccAddress, balances sdk.C
 
 // SignAndDeliver signs and delivers a transaction. No simulation occurs as the
 // ibc testing package causes checkState and deliverState to diverge in block time.
+//
+// CONTRACT: BeginBlock must be called before this function.
 func SignAndDeliver(
```

Comment:
SignAndDeliver will no longer simulate a block internally. It just calls DeliverTx and assumes BeginBlock has been called earlier